### PR TITLE
Custom hooked `Map` externalisation

### PR DESCRIPTION
**Pull request by geo2a**
Work on  #321

Just adding integration tests for now.

~I'm struggling to reproduce #321 as the provided bug report unfortunately contains a pre-multy-or LLVM definition. I've requested access to the relevant code bases in order to produce obtain the up-to-date definition.~

This turns out to be an issue in the LLVM backend; hence, blocked on https://github.com/runtimeverification/llvm-backend/issues/886

### CODE REVIEW

File: test/llvm-integration/LLVM.hs [hunk 1]
```diff
@@ -1301,6 +1380,42 @@ defSymbols =
                         }
                 }
             )
+        ,
+            ( "Lbl'Unds'Int2Int'Pipe'-'-GT-Unds'"
+            , Symbol
+                { name = "Lbl'Unds'Int2Int'Pipe'-'-GT-Unds'"
+                , sortVars = []
+                , argSorts = [SortApp "SortWrappedInt" [], SortApp "SortInt" []]
```
Comment (Baltoli) [inline test/llvm-integration/LLVM.hs:1388]: @geo2a I don't follow why the second argument has sort `Int` here; in the corresponding definition the second argument has sort `WrappedInt`? I might be missing something though.

File: test/llvm-integration/definition/llvm.k [hunk 2]
```diff
@@ -39,3 +40,30 @@ module LLVM
...
+  syntax MapInt2Int ::= MapInt2Int MapInt2Int
+         [ left, function, hook(MAP.concat), klabel(_MapInt2Int_),
+           symbol, assoc, comm, unit(.MapInt2Int), element(_Int2Int|->_),
+           index(0), format(%1%n%2)
+         ]
+  syntax MapInt2Int ::= ".MapInt2Int"
+         [ function, total, hook(MAP.unit),
+           klabel(.MapInt2Int), symbol, latex(\dotCt{MapInt2Int})
+         ]
+  syntax MapInt2Int ::= WrappedInt "Int2Int|->" WrappedInt
```
Comment (Baltoli) [inline test/llvm-integration/definition/llvm.k:63]: Here's the production I believe corresponds to the above comment @geo2a

File: test/llvm-integration/LLVM.hs (hunk 1 above)
Comment (geo2a) [reply test/llvm-integration/LLVM.hs:1388]: You're right, and it indeed makes the test I constructed to pass. We have been able to pin-point the problem to the `MAP.update` hook in the LLVM backend. I'll update the integration tests and also open an issue in the LLVm backend with the somewhat minimized definition and input.

Comment (Baltoli) [reply test/llvm-integration/LLVM.hs:1388]: Remarkable timing; I had also just managed to build and get the same test to pass with the value sort as `WrappedInt`! Let's take a look at your new minimized example in our meeting this afternoon.

**Comment by goodlyrottenapple** [comment]
closing in favour of #378

**Pull request closed without merging**